diff --git a/Cargo.lock b/Cargo.lock
index f4b4032..422d478 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -3016,6 +3016,7 @@ dependencies = [
  "librespot-core",
  "librespot-discovery",
  "librespot-playback",
+ "librespot-protocol",
  "log",
  "pledge",
  "rspotify",
diff --git a/Cargo.toml b/Cargo.toml
index 71a3cd0..4491562 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,6 +33,7 @@ librespot-playback = { version = "0.4", default-features = false }
 librespot-core = { version = "0.4" }
 librespot-discovery = { version = "0.4" }
 librespot-connect = { version = "0.4" }
+librespot-protocol = { version = "0.4" }
 toml = "0.7"
 color-eyre = "0.6"
 directories = "5.0.1"
diff --git a/src/setup.rs b/src/setup.rs
index 07a713c..5ac5e8b 100644
--- a/src/setup.rs
+++ b/src/setup.rs
@@ -16,6 +16,7 @@ use librespot_playback::{
 #[allow(unused_imports)] // cfg
 use log::{debug, error, info, warn};
 use std::{str::FromStr, thread, time::Duration};
+use librespot_protocol::authentication::AuthenticationType;
 
 pub(crate) fn initial_state(config: config::SpotifydConfig) -> main_loop::MainLoop {
     let mixer = {
@@ -160,10 +161,14 @@ fn get_credentials(
         }
     }
 
-    Some(Credentials::with_password(
+    let mut credentials = Credentials::with_password(
         username.as_ref()?,
         password.as_ref()?,
-    ))
+    );
+
+    credentials.auth_type = AuthenticationType::AUTHENTICATION_SPOTIFY_TOKEN;
+
+    Some(credentials)
 }
 
 fn find_backend(name: Option<&str>) -> fn(Option<String>, AudioFormat) -> Box<dyn Sink> {
